# HG changeset patch
# User Sukhbir Singh <sukhbir@torproject.org>
# Date 1413963006 -19800
#      Wed Oct 22 13:00:06 2014 +0530
# Node ID 1d142c56cdf63cb3b0bcab81607c72c4b8a04599
# Parent  ce691e5400583295a0cc0d0d43a3382a2f95940f
Set default Instantbird preferences with enhanced security settings

diff --git a/im/app/profile/all-instantbird.js b/im/app/profile/all-instantbird.js
--- a/im/app/profile/all-instantbird.js
+++ b/im/app/profile/all-instantbird.js
@@ -8,8 +8,6 @@
 pref("browser.hiddenWindowChromeURL", "chrome://instantbird/content/hiddenWindow.xul");
 #endif
 
-#expand pref("general.useragent.extra.instantbird", "Instantbird/__APP_VERSION__");
-
 #ifdef XP_UNIX
 #ifndef XP_MACOSX
 #define UNIX_BUT_NOT_MAC
@@ -244,11 +242,6 @@
 pref("extensions.getMoreEmoticonsURL", "https://add-ons.instantbird.org/%LOCALE%/%APP%/%VERSION%/emoticons/");
 pref("extensions.getMoreProtocolsURL", "https://add-ons.instantbird.org/%LOCALE%/%APP%/%VERSION%/protocols/");
 
-// suppress external-load warning for standard browser schemes
-pref("network.protocol-handler.warn-external.http", false);
-pref("network.protocol-handler.warn-external.https", false);
-pref("network.protocol-handler.warn-external.ftp", false);
-
 // don't load links inside Instantbird
 pref("network.protocol-handler.expose-all", false);
 // Although we allow these to be exposed internally, there are various places
@@ -302,14 +295,61 @@
 // 3  at the end of the tabstrip
 pref("browser.tabs.closeButtons", 1);
 
-#expand pref("chat.irc.defaultQuitMessage", "Instantbird __APP_VERSION__ -- http://www.instantbird.com");
-
 pref("chat.twitter.consumerKey", "TSuyS1ieRAkB3qWv8yyEw");
 pref("chat.twitter.consumerSecret", "DKtKaSf5a7pBNhdBsSZHTnI5Y03hRlPFYWmb4xXBlkU");
 
 // Comma separated list of prpl ids that should use libpurple even if there is
 // a JS implementation. This is used to land JS-prpls pref'ed off in nightlies.
-pref("chat.prpls.forcePurple", "prpl-jabber");
+pref("chat.prpls.forcePurple", "");
 
 // Whether to parse log files for conversation statistics.
 pref("statsService.parseLogsForStats", true);
+
+/* Tor Messenger */
+// Logging
+// Disable all logging
+pref("purple.logging.format", "json");
+pref("purple.logging.log_chats", false);
+pref("purple.logging.log_ims", false);
+pref("purple.logging.log_system", false);
+
+// Network
+// Use a manual proxy configuration
+pref("network.proxy.type", 1);
+// Configure Instantbird to use the SOCKS5 proxy
+pref("network.proxy.socks", "127.0.0.1");
+pref("network.proxy.socks_port", 9150);
+pref("network.proxy.socks_version", 5);
+// Set DNS proxying through SOCKS5
+pref("network.proxy.socks_remote_dns", true);
+// Warn when an external application is to be launched
+pref("network.protocol-handler.warn-external.http", true);
+pref("network.protocol-handler.warn-external.https", true);
+pref("network.protocol-handler.warn-external.ftp", true);
+pref("network.protocol-handler.warn-external.file", true);
+pref("network.protocol-handler.warn-external-default", true);
+
+// Security
+// Disable SSLv3 by setting the minimum supported protocol to TLS 1.0.
+pref("security.tls.version.min", 1);
+// Disable geolocation
+pref("geo.enabled", false);
+// Empty the user agent
+pref("general.useragent.extra.instantbird", "");
+
+// Messenger
+//  0 = do not connect / show the account manager
+pref("messenger.startup.action", 0);
+// Do not report idle status or the away message
+pref("messenger.status.awayWhenIdle", false);
+pref("messenger.status.defaultIdleAwayMessage", "");
+pref("messenger.status.reportIdle", false);
+
+// Chat
+// Do not show any quit message
+pref("chat.irc.defaultQuitMessage", "");
+
+// Browser
+// Disable caching
+pref("browser.cache.disk.enable", false);
+pref("browser.cache.offline.enable", false);
+// Disable text formatting (remove the tags)
+pref("messenger.options.filterMode", 0);
+
+// Navigator
+// Disable "Take Picture" functionality that accesses the webcam
+pref("media.navigator.video.enabled", false);
